src/test/java/com/yelp/nrtsearch/server/luceneserver/field/DateTimeFieldDefTest.java

```diff
@@ -0,0 +1,200 @@
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+public class DateTimeFieldDefTest extends ServerTestCase {
+
+  @ClassRule public static final GrpcCleanupRule grpcCleanup = new GrpcCleanupRule();
+
+  protected List<String> getIndices() {
+    return Collections.singletonList(DEFAULT_TEST_INDEX);
+  }
+
+  protected FieldDefRequest getIndexDef(String name) throws IOException {
+    return getFieldsFromResourceFile("/field/registerFieldsDateTime.json");
+  }
+
+  protected void initIndex(String name) throws Exception {
+    List<AddDocumentRequest> docs = new ArrayList<>();
+    AddDocumentRequest docWithTimestamp1 =
+        AddDocumentRequest.newBuilder()
+            .setIndexName(name)
+            .putFields("doc_id", MultiValuedField.newBuilder().addValue("1").build())
+            .putFields(
+                "timestamp_epoch_millis",
+                MultiValuedField.newBuilder().addValue("1611742000").build())
+            .putFields(
+                "timestamp_string_format",
+                MultiValuedField.newBuilder().addValue("2021-01-27 10:06:40").build())
+            .build();
+    AddDocumentRequest docWithTimestamp2 =
+        AddDocumentRequest.newBuilder()
+            .setIndexName(name)
+            .putFields("doc_id", MultiValuedField.newBuilder().addValue("2").build())
+            .putFields(
+                "timestamp_epoch_millis",
+                MultiValuedField.newBuilder().addValue("1610742000").build())
+            .putFields(
+                "timestamp_string_format",
+                MultiValuedField.newBuilder().addValue("2021-01-15 20:20:00").build())
+            .build();
+    docs.add(docWithTimestamp1);
+    docs.add(docWithTimestamp2);
+    addDocuments(docs.stream());
+  }
+
+  @Test
+  public void testDateTimeRangeQueryEpochMillis() {
+    SearchResponse response =
+        doQuery(
+            Query.newBuilder()
+                .setRangeQuery(
+                    RangeQuery.newBuilder()
+                        .setField("timestamp_epoch_millis")
+                        .setLower("1610741000")
+                        .setUpper("1610743000")
+                        .build())
+                .build(),
+            List.of("doc_id"));
+    assertFields(response, "2");
+  }
+
+  @Test
+  public void testDateTimeRangeQueryStringDateFormat() {
+    SearchResponse response =
+        doQuery(
+            Query.newBuilder()
+                .setRangeQuery(
+                    RangeQuery.newBuilder()
+                        .setField("timestamp_string_format")
+                        .setLower("2021-01-27 10:05:40")
+                        .setUpper("2021-01-27 10:07:40")
+                        .build())
+                .build(),
+            List.of("doc_id"));
+    assertFields(response, "1");
+  }
+
+  @Test(expected = RuntimeException.class)
```

Comment:
Is there a less general exception we can test? Every unchecked exception is going to be a RuntimeException. You can also catch the exception and assert the message like in https://github.com/Yelp/nrtsearch/blob/master/src/test/java/com/yelp/nrtsearch/server/grpc/LuceneServerTest.java#L952 which is better when an exception specific to your error is not thrown

Reply:
Decided to assert for the exception message